### PR TITLE
fix(bot): scaffolding bring last project's context

### PR DESCRIPTION
**Pull request by YitongFeng-git**
fix bug https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10430876

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/bot/plugin.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export class TeamsBotImpl {
 
   public async preScaffold(context: PluginContext): Promise<FxResult> {
     this.ctx = context;
-    await this.config.restoreConfigFromContext(context);
+    this.config = new TeamsBotConfig();
```
Comment (henzhang-ms) [inline packages/fx-core/src/plugins/resource/bot/plugin.ts:86]: does solution passes the config of current project to preScaffold of new project?

Comment (YitongFeng-git) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:86]: It is not related with any restore method, after we run local debug, the settings will store in TeamsBotImpl.localDebugConfig until we reopen a folder or restart vscode. Another solution is let the TeamsBotImpl not be global.
image

Comment (eriolchan) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:86]: @henzhang-ms @hund030 will this issue also happen to Frontend and Function plugin?

According to the code of [teamsBotConfig.ts](https://github.com/OfficeDev/TeamsFx/blob/a0787e0cf142f1cb1c9ee58aa6592dbdea48aeed/packages/fx-core/src/plugins/resource/bot/configs/teamsBotConfig.ts#L26), why do we always restore configs of all stages (scaffold, provision, deploy, local-debug) even when toolkit is executing only one stage of them? @IvanJobs do you have more information and comments about the fix?

Comment (henzhang-ms) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:86]: Should no for function, function all the config as context given, that's why it is called `syncConfigFromContext`. I think the root cause of bot is that it only sync from context if the context value is not empty. But I am not sure whether it has some by-design concern.

```
    const objectIdValue: ConfigValue = context.config.get(PluginBot.OBJECT_ID);
    if (objectIdValue) {
      this.objectId = objectIdValue as string;
    }
```

Comment (hund030) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:86]: No for frontend, either. Frontend plugin is totally stateless now. It only accepts input from context.

**Pull request closed without merging**